fix(ci): 修复 claude-code-review 工作流 - 使用 ubuntu runner

## 问题描述

Claude Code Review 工作流在所有 PR 上失败，错误信息：

```
Windows is not supported
##[error]Process completed with exit code 1.
```

## 根本原因

`.github/workflows/claude-code-review.yml` 第 21 行配置了错误的 runner：

```yaml
runs-on: windows-latest  # ❌ Claude Code Action 不支持 Windows
```

Claude Code Action (`anthropics/claude-code-action@beta`) 仅支持 Linux runner。

## 解决方案

将 runner 改为 `ubuntu-latest`：

```yaml
runs-on: ubuntu-latest  # ✅ Claude Code Action 需要 Linux runner
```

## 验收标准

- [ ] [CI-919] 修改 `.github/workflows/claude-code-review.yml` 的 `runs-on` 为 `ubuntu-latest`
- [ ] [CI-919] 创建测试 PR 验证 claude-review 检查能成功运行
- [ ] [CI-919] 确认 claude-review 能在 PR 上添加审查评论

## 影响范围

所有 Pull Request 的自动化 code review 功能当前不可用。

## 优先级

P2 - 影响开发体验但不阻塞合并流程（可通过人工审查替代）

## 参考

- Claude Code Action: https://github.com/anthropics/claude-code-action
- 失败示例：PR #918 的 claude-review 检查

--- a/.github/workflows/claude-code-review.yml
+++ b/.github/workflows/claude-code-review.yml
@@ -17,8 +17,8 @@ jobs:
     #   github.event.pull_request.user.login == 'external-contributor' ||
     #   github.event.pull_request.user.login == 'new-developer' ||
     #   github.event.pull_request.author_association == 'FIRST_TIME_CONTRIBUTOR'
-    
-    runs-on: windows-latest
+
+    runs-on: ubuntu-latest
     permissions:
       contents: read
       pull-requests: read

